plugin/plugin.go

```diff
@@ -423,6 +428,88 @@ func wrap_log_callback(sev C.int, msg *C.char, ud *C.user_data_t) C.int {
 	return 0
 }
 
+type configValueType int
+
+const (
+	configTypeString configValueType = iota
+	configTypeNumber
+	configTypeBoolean
+)
+
+// ConfigValue is C.oconfig_value_t implemented as a Go struct.
+type ConfigValue struct {
+	typ configValueType
+	s   string
+	f   float64
+	b   bool
+}
+
+func (v ConfigValue) String() (string, bool) {
+	return v.s, v.typ == configTypeString
+}
+
+func (v ConfigValue) Number() (float64, bool) {
+	return v.f, v.typ == configTypeNumber
+}
+
+func (v ConfigValue) Boolean() (bool, bool) {
+	return v.b, v.typ == configTypeBoolean
+}
+
+// Config is C.oconfig_item_t in Go form, with convenience functions.
```

Comment:
This leaks implementation details. I think it would be more useful to say along the lines of: "Config represents one configuration block, which may contain other configuration blocks."